File: src/com/leozhi/topic701/Solution03.java
package com.leozhi.topic701;

/**
 * @author leozhi
 * 递归
 * 通过
 * 0ms
 */
public class Solution03 {
    public TreeNode insertIntoBST(TreeNode root, int val) {
        if (root == null) {
            return new TreeNode(val);
        }

        if (val < root.val) {
            root.left = insertIntoBST(root.left, val);
        } else {
            root.right = insertIntoBST(root.right, val);
        }
        return root;
    }
}

File: src/com/leozhi/topic222/Solution03.kt
package com.leozhi.topic222

/**
 * @author leozhi
 * 递归
 * 通过
 * 252ms
 */
class Solution03 {
    private fun countNodes(root: TreeNode?): Int {
        val res = 0
        if (root == null) {
            return res
        }
        val left = countNodes(root.left)
        val right = countNodes(root.right)
        return 1 + left + right
    }
}